### ISSUE TITLE
Improve snippet error readability for every day users

### ISSUE THREAD

**Issue by fidgetingbits** [cursorless-dev/cursorless#1928]
Just a suggestion, but I suspect the errors you get when using using snippets are probably quite confusing to people not familiar with cursorless internals.

For example if I just have a python line `foo = 1` and I want to test out the `class` snippet from pokey's repo, so say `class wrap fin`, I'm greeted with the error `Couldn't find containing namedFunction.`. I happen to know what this means because I've been adding the nix language to cursorless lately, but prior to that it probably wouldn't have been so clear.

I wonder if there's a way to associate these scope types with some human understandable information to put into errors, so that it would say: `This snippet can only wrap a function, but couldn't find containing namedFunction scope.`

This at least gives the user the opportunity to keep trying stuff, rather than having to immediately dig into the actual `.cursorless-snippet` to figure out what it's doing, and/or potentially read up on what `namedFunction` actually means.

There is of course the risk that this will become too long/verbose, which would then be annoying during development.

**Comment by pokey** [comment]
Makes sense. Fwiw it is possible to wrap something that's not a function: you need to give it a target with an explicit type. Eg "class wrap token fin" or "class wrap just fin". Or "class wrap line fin" if you want the line. It just expands to function if it gets a "weak" target

But agreed a better error message would be nice
